terraform/modules/upload-service/checksum_daemon.tf

```diff
@@ -0,0 +1,94 @@
+resource "aws_iam_role" "checksum_lambda" {
+  name = "checksum-${var.deployment_stage}"
+  assume_role_policy = <<POLICY
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Sid": "",
+      "Effect": "Allow",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Action": "sts:AssumeRole"
+    }
+  ]
+}
+POLICY
+}
+
+resource "aws_iam_role_policy" "checksum_lambda" {
+  name = "checksum-${var.deployment_stage}"
+  role = "${aws_iam_role.checksum_lambda.name}"
+
+    policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Effect": "Allow",
+      "Action": [
+        "logs:CreateLogGroup",
+        "logs:CreateLogStream",
+        "logs:PutLogEvents"
+      ],
+      "Resource": "arn:aws:logs:*:*:*"
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "sqs:ChangeMessageVisibility",
+        "sqs:DeleteMessage",
+        "sqs:GetQueueAttributes",
+        "sqs:ReceiveMessage"
+      ],
+      "Resource": [
+        "arn:aws:sqs:*:*:${aws_sqs_queue.upload_queue.name}"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+        "secretsmanager:DescribeSecret",
+        "secretsmanager:GetSecretValue"
+      ],
+      "Resource": [
+        "arn:aws:secretsmanager:us-east-1:${local.account_id}:secret:dcp/upload/${var.deployment_stage}/*"
+      ]
+    },
+    {
+      "Effect": "Allow",
+      "Action": [
+          "s3:GetObject",
+          "s3:GetObjectTagging",
+          "s3:PutObjectTagging"
+      ],
+      "Resource": [
+          "arn:aws:s3:::${aws_s3_bucket.upload_areas_bucket.bucket}/*"
+      ]
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_lambda_function" "checksum_lambda" {
+  function_name    = "checksum-${var.deployment_stage}"
+  s3_bucket        = "checksum-lambda"
```

Comment:
Is this an actual S3 bucket name?  If so to avoid naming conflicts we should prefix all HCA buckets with `org-humancellatlas-`